### PR TITLE
Add base extended ACL support.

### LINKED ISSUE #108: Extended ACLs missing
Extended ACLs are not supported. Will they ever be?

### PR DESCRIPTION
**Pull request by mharista**
Fixes #108

### CODE REVIEW

**Comment by mharista** [comment]
I'm interested in feedback on the level of extended ACLs to support. Commands can get pretty detailed and complex. I'll continue adding to this but trying to support everything may be excessive. For example below is a valid extended ACL configuration.

ip access-list exttest
   10 deny ip any any log
   20 permit tcp any eq http-rpc-epmap https z39-50 2.2.2.0/24 ack fin psh rst syn urg ttl eq 2 fragments tracked log

File: test/unit/test_api_acl.py [hunk 1]
```diff
@@ -91,23 +157,107 @@ def test_update_entry(self):
...
+        self.eapi_positive_config_test(func, cmds)
+
+    def test_add_entry_with_seqno(self):
+        cmds = ['ip access-list exttest',
+                '30 permit ip 0.0.0.0/32 1.1.1.1/32 log', 'exit']
+        func = function('add_entry', 'exttest', 'permit', 'ip', '0.0.0.0',
+                        '32', '1.1.1.1', '32', True, 30)
+        self.eapi_positive_config_test(func, cmds)
+
+
```
Comment (dathelen) [inline test/unit/test_api_acl.py]: might be good idea to add a test case with 'host' keyword since the parsing will be a bit different

**Comment by dathelen** [comment]
Also, FWIW, I agree we dont have to support every permutation.  For example matching on TCP flags isn't something that most of HW supports anyways so we can probably skip that.  I do think the host keyword is a common use case in my experience.

File: pyeapi/api/acl.py [hunk 2]
```diff
@@ -152,5 +215,90 @@ def remove_entry(self, name, seqno):
...
+            entry['srcaddr'] = match.group(6) or 'any'
+            entry['srclen'] = match.group(7)
+            entry['srcport'] = match.group(8)
+            entry['dstaddr'] = match.group(11) or 'any'
+            entry['dstlen'] = match.group(12)
+            entry['dstport'] = match.group(14)
+            entry['other'] = match.group(15)
+            entries[match.group(1)] = entry
+        return dict(entries=entries)
+
```
Comment (mharista) [inline pyeapi/api/acl.py]: Good point. I knew there was some cleaning up I would need to do in this area.

File: pyeapi/api/acl.py [hunk 3]
```diff
@@ -152,5 +215,90 @@ def remove_entry(self, name, seqno):
...
+    def default(self, name):
+        return self.configure('default ip access-list %s' % name)
+
+    def update_entry(self, name, seqno, action, protocol, srcaddr,
+                     srcprefixlen, dstaddr, dstprefixlen, log=False):
+        cmds = ['ip access-list %s' % name]
+        cmds.append('no %s' % seqno)
+        entry = '%s %s %s %s/%s %s/%s' % (seqno, action, protocol, srcaddr,
+                                          srcprefixlen, dstaddr, dstprefixlen)
```
Comment (mharista) [inline pyeapi/api/acl.py]: Currently I've remained consistent with how standard ACLs are handling the host keyword for updated/new entries. The user is required to put in 0 for the host prefix length and EOS will translate it to the host keyword.

**Pull request merged**